### ISSUE TITLE
How to pass a shared handle to D3DImage in a managed way?

### ISSUE THREAD

**Issue by essoperagma** [dotnet/wpf#6141]
I have a Win32 application that is rendering a 3D scene onto a **shared texture** using Direct3D 9ex.

I also have a WPF application, running as a **separate process**. It has access to the shared handle of the texture previously created by the 3D application.

To be able to display this image in my WPF app, I am trying to utilize 3D3Image.SetBackBuffer, which requires a pointer to the texture.
```
d3dimg.SetBackBuffer(D3DResourceType.IDirect3DSurface9, pSurface);
```

However, at this point, I only have a shared handle and not a texture. My questions are:

1. How can I create a texture from the shared handle while staying in managed code? Even an internal method that I can call via reflection would be acceptable for me.
2. If this isn't possible, is there an unmanaged alternative that I can use with PInvoke? I'm fine with calling [IDirect3DDevice9::CreateTexture](https://docs.microsoft.com/en-us/windows/win32/api/d3d9/nf-d3d9-idirect3ddevice9-createtexture) using PInvoke, but that requires me to have a pointer to the D3DDevice which I don't have. Can I somehow access the device created by WPF runtime? If I created another device myself, can a texture created by this device be used with 3D3Image?

**Comment by essoperagma** [comment]
I have managed to get this done using an unmanaged dll and calling the following functions:
- [CreateWindow](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-createwindowexa)
- Direct3DCreate9Ex
- IDirect3D9Ex::CreateDeviceEx
- IDirect3DDevice9Ex::CreateTexture
- IDirect3DSurface9::GetSurfaceLevel

Even though I'm unblocked at the moment, I think it would be nice to have this functionality included publicly in D3DImage type for convenience (assuming there isn't already an alternative managed way). This would eliminate a great deal of extra work such as:
- Creating the unmanaged project & setting it up in CI/CD
- Handling the platform (different DLL files for x86 and x64)
- Creating a temporary window,
- Creating Direct3D9Ex and IDirect3DDevice9Ex objects while they already exist somewhere in WPF runtime.

I'm updating the title to remove the [Question] tag. I believe this is more of a "feature request" now.

**Comment by lindexi** [comment]
@essoperagma Could you try my demo? 

https://github.com/lindexi/lindexi_gd/tree/f4c2f884b3fb006676aeef7e249055c5e2d8766d/RawluharkewalQeaninanel

**Comment by essoperagma** [comment]
Thanks @lindexi. It seems that you have used a 3rd party package to access D3D functionality. I guess that is another way to unblock the development.

I'm wondering if this could be included in `D3DImage` type so we wouldn't need these relatively more difficult ways.
